src/databricks/labs/ucx/hive_metastore/locations.py

```diff
@@ -530,3 +552,23 @@ def _is_json(self, file_name: str) -> bool:
 
     def _is_irrelevant(self, file_name: str) -> bool:
         return any(pattern in file_name for pattern in self._fiter_paths)
+
+    def _is_streaming_checkpoint_dir(self, file_info: FileInfo) -> bool:
+        path_dirs = [file.name for file in self._dbutils.fs.ls(file_info.path) if file.size == 0]
+        return 'commits/' in path_dirs and 'offsets/' in path_dirs
+
+    def _is_recursible_dir(self, file_info: FileInfo) -> bool:
+        # Rules for recursing into a folder
+        # - should be size 0 (usually a directory)
+        # - should not be the _delta_log directory
+        # - file should not be partitioned
+        # - file name should not start with part-
+        # - brackets are not allowed in dbutils.fa.ls
```

Comment:
Does this make a directory non-recursible or should it be handled when calling the `ls`?

```suggestion
        # - brackets are not allowed in dbutils.fs.ls
```

Reply:
yes- dbutils cannot handle brackets. Something I encountered in the wild during testing on a customer's mount directory.